Gazel-Tutorial/src/Gazel.Tutorial.Module.ProductManagement/ProductManager.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Gazel;
+using Gazel.Tutorial.Module.ProductManagement.Service;
+
+namespace Gazel.Tutorial.Module.ProductManagement
+{
+    public class ProductManager : IProductManagerService
+    {
+        private readonly IModuleContext context;
+
+        public ProductManager(IModuleContext context)
+        {
+            this.context = context;
+        }
+
+        public Product CreateProduct(string name, float price, int stock)
+        {
+            return context.New<Product>().With(name, price, stock);
+        }
+
+        public void DeleteProduct(Product product)
+        {
+            context.Query<CartItems>().ByProduct(product).ForEach(t => t.Cart.RemoveFromCart(product));
+
+            product.RemoveProduct();
```

Comment:
bu logic product'in kendi icinde olabilir, urun kendini silerken once altindaki cartitem'lari silebilir

Reply:
Düzelttim.